src/ios/CDVCamera.m

```diff
@@ -145,9 +201,38 @@ - (void)takePicture:(CDVInvokedUrlCommand*)command
     [self.commandDelegate runInBackground:^{
         
         CDVPictureOptions* pictureOptions = [CDVPictureOptions createFromTakePictureArguments:command];
+        
+        /*
+         FIXME #1
+         What to do about quality?
+         If the option is set and the image is retrieved from the PhotoLibrary or SavedAlbum with a NATIVE_URI, no editing,
+         and is not saved to the photo album, then we return the "wrong" image.
+         However, the doc says "quality of the **saved** image", so it might be fine.
+         */
+        /*
+         FIXME #2
+         The doc says: "Rotate the image to correct for the orientation of the device **during capture**."
+         Is capture <=> sourceType == CAMERA ?
+         */
+        // Check for option compatibility
+        BOOL isDestinationNativeUri = (pictureOptions.destinationType == DestinationTypeNativeUri);
+        
+        BOOL needsResize = [self needsResize:pictureOptions];
+        BOOL needsOrientationCorrection = pictureOptions.correctOrientation;
+        BOOL isSourceCamera = (pictureOptions.sourceType == UIImagePickerControllerSourceTypeCamera);
+        BOOL allowsEditing = pictureOptions.allowsEditing;
+        BOOL needsSavingToPhotoAlbum = (needsResize || needsOrientationCorrection || allowsEditing);
+        
+        // if one wants an edited image and a NATIVE_URI, the edited image must be in the assets library therefore one must set the saveToPhotoAlbum option to true.
+        if (!pictureOptions.saveToPhotoAlbum && isDestinationNativeUri && needsSavingToPhotoAlbum) {
```

Comment:
Hmmm, I agree that we should ideally not save into the photoAlbum if the user hasn't specified it in the options.
But on the other hand, the current behavior is to edit the image anyway, then go ahead and save it in the assets library/saved photo album anyway.
Which one should we go with ? I'm not really sure.